kubetest/main.go

```diff
@@ -512,6 +509,16 @@ func prepareGcp(o *options) error {
 	if err := migrateGcpEnvAndOptions(o); err != nil {
 		return err
 	}
+	if o.provider == "gke" && o.gcpNodeImage != "" {
+		// TODO(kubernetes/test-infra#3307): Should disappear when
+		// "bash" deployment is gone.
+		if o.deployment == "bash" {
+			os.Setenv("KUBE_GKE_IMAGE_TYPE", o.gcpNodeImage)
+		}
+		// TODO(kubernetes/test-infra#3536): This is used by the
+		// ginkgo-e2e.sh wrapper.
+		os.Setenv("NODE_OS_DISTRIBUTION", o.gcpNodeImage)
```

Comment:
`KUBE_NODE_OS_DISTRUBTION` should be set by the `migrateOptions` above. However, `NODE_OS_DISTRUBTION` is what's expected in the wrapper. This assignment is done in GCE, but not in GKE, even in the bash distro:
* GCE: https://github.com/kubernetes/kubernetes/blob/ab40f526fd125a354859ca187fede8591bf25d0a/cluster/gce/config-default.sh#L49 
* GKE (trying to prove the negative): https://github.com/kubernetes/kubernetes/search?utf8=%E2%9C%93&q=NODE_OS_DISTRIBUTION&type=